src/cli/subcommands/snapshot_cmd.rs

```diff
@@ -142,20 +158,92 @@ impl SnapshotCommands {
             Self::Validate {
                 recent_stateroots,
                 snapshot,
-            } => validate(&config, recent_stateroots, snapshot).await,
+            } => {
+                // this is all blocking...
+                use std::fs::File;
+                match CompressedCarV1BackedBlockstore::new(BufReader::new(File::open(snapshot)?)) {
+                    Ok(store) => {
+                        validate_with_blockstore(
+                            &config,
+                            store.roots(),
+                            Arc::new(store),
+                            recent_stateroots,
+                        )
+                        .await
+                    }
+                    Err(error)
+                        if error.kind() == std::io::ErrorKind::Other
+                            && error.get_ref().is_some_and(|inner| {
+                                inner.downcast_ref::<MaxFrameSizeExceeded>().is_some()
+                            }) =>
+                    {
+                        bail!("The provided compressed car file cannot be used as a blockstore. Prepare it using `forest snapshot compress ...`")
+                    }
+                    Err(error) => {
+                        info!(%error, "file may be uncompressed, retrying as a plain CAR...");
+                        let store = UncompressedCarV1BackedBlockstore::new(File::open(snapshot)?)?;
+                        validate_with_blockstore(
+                            &config,
+                            store.roots(),
+                            Arc::new(store),
+                            recent_stateroots,
+                        )
+                        .await
+                    }
+                }
+            }
+            Self::Compress {
+                source,
+                destination,
+                compression_level,
+                frame_size,
+            } => {
+                // We've got a binary blob, and we're not exactly sure if it's compressed, and we can't just peek the header:
+                // For example, the zstsd magic bytes are a valid varint frame prefix:
+                assert_eq!(
+                    unsigned_varint::io::read_usize(&[0xFD, 0x2F, 0xB5, 0x28][..]).unwrap(),
+                    6141,
+                );
```

Comment:
Does it belong in the runtime? It seems like something we could check at compile time, though this `read_usize` probably is not `const`. Perhaps it would belong more in tests if anywhere?

Reply:
The purpose of the assertion is to illustrate why we can't easily tell whether an archive is compressed or not.